### ISSUE TITLE
4.7 Update compile time cannot find namespace _esri

### ISSUE THREAD

**Issue by jcarasick** [Esri/esri-loader#109]
i've upgraded recently and am getting compile time errors stating cannot find the _esri namespace. I've tried a few things and can't seem to resolve. Note the project does seem to run and i can create a map, etc.

i've tried a few different ways to import but none seemed to work.  Any ideas?

import { loadModules } from 'esri-loader';
import esri = __esri;
//import * as esri from 'arcgis-js-api';
//import esri from require('arcgis-js-api');

**Comment by jwasilgeo** [comment]
`__esri` is the namespace for the TypeScript typings for the ArcGIS API for JavaScript v4 and shouldn't have anything to do with `esri-loader`.

Please make sure those are installed and up to date if you are going to be importing and using those typings.

https://github.com/Esri/jsapi-resources/tree/master/4.x/typescript

**Comment by jcarasick** [comment]
I agree but I rolled back to the 2.2.0 esri-loaded and the errors went away. any ideas on that one?

**Comment by andygup** [comment]
What framework are you using, e.g. React, Angular, Vue, etc?

**Comment by jwasilgeo** [comment]
Curious!

**Comment by jcarasick** [comment]
Angular 5

**Comment by jcarasick** [comment]
it's weird, there are no errors in the IDE at all, but when i run ng serve i get compile errors that it can't find namespace _esri. however I can browse to the web map

**Comment by tomwayson** [comment]
> I agree but I rolled back to the 2.2.0 esri-loaded and the errors went away. any ideas on that one?

Unless you are explicitly passing a URL, esri-loader v2.2 defaults to JSAPI 4.6, 2.3 defaults to 4.7. Is the version of your types is aligned with the version of the API that is loaded by esri-loader?

Other than that, I don't see anything in [the diff](https://github.com/Esri/esri-loader/compare/v2.2.0...v2.3.0) that is relevant to this issue.

**Comment by jcarasick** [comment]
I ended up recreating the project. I did try to blow away the node_modules folder. I was using the 4.7 with the 2.2 loader ok passing that url as an option. After rebuilding the project and bringing in the esr-loader 2.3 i did not get the same issues. I also noticed i was using the newer version of the angular cli. not sure if there was something to that or not as well. at any rate i have this new loader working without error so thanks.
